.github/workflows/main.yml

```diff
@@ -1,42 +1,188 @@
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            /home/runner/.cache/go-build
+            ~/Library/Caches/go-build
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Run tests
+        run: |
+          make tests
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check go.sum
+        run: |
+          cat go.sum
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+
+  build:
+    name: Build
+    needs: [test] 
+    runs-on: ubuntu-20.04
+    env: #TODO change DOCKERTOKEN and DOCKERUSERNAME
+      dockerToken: ${{ secrets.DOCKERTOKEN }}
+      dockerUsername: ${{ secrets.DOCKERUSERNAME }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Start cluster for e2e tests
+        run: |
+          ./e2e/setup/setup_cluster_ci.sh
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Build & Run e2e tests
+        run: | 
+          make build-ci tests-e2e 
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check go.sum
+        run: |
+          cat go.sum
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            /home/runner/.cache/go-build
+            ~/Library/Caches/go-build
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: armadactl
+          path: |
+            ${{ env.GOPATH }}/src/github.com/G-Research/armada/bin/armadactl
+
+      - name: Push Image #TODO
+        run: |
+          if [ -z "${dockerUsername}" ]
+            then
+              echo "Do not push image inside fork."
+              exit 0
+          fi
+
+          TAG=${GITHUB_SHA}
+          BRANCH=${GITHUB_REF##*/}
+
+          if [ ${BRANCH} != master ]
+            then
+              TAG=branch-$(echo -n ${BRANCH} | sed 's|/|-|g')-${GITHUB_SHA}
+          fi
+

```

Comment:
You also need to push armada-binoculars too now. See:
https://github.com/G-Research/armada/blob/master/.circleci/config.yml#L247

Reply:
@JamesMurkin added armada-binoculars: [L193 ](https://github.com/G-Research/armada/pull/597/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R193)

FYI I'm pushing helm charts for armada [here ](https://github.com/ljubon/charts/blob/test-dispatcher/.github/workflows/gh-pages.yml#L58)with `workflow_dispacher` which is implemented [here](https://github.com/G-Research/armada/pull/597/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R123)
so releasing new charts process looks like this...
- new tag pushed to `ljubon/armada`  is triggering[ release workflow](https://github.com/ljubon/armada/blob/master/.github/workflows/release.yml#L90) on armada repo
- result of that job new GH Actions is triggered on [ljubon/chart](https://github.com/G-Research/charts/pull/39) (PR part1) repo which is publishing charts directly to gh-pages ([e.g of job](https://github.com/ljubon/charts/runs/3285250244?check_suite_focus=true))

e.g [result](https://github.com/ljubon/charts/tree/test-dispatcher/armada), binoculars are present now

pretty simple 😄 💣